osquery/tables/system/windows/users.cpp

```diff
@@ -65,6 +65,46 @@ std::string getUserHomeDir(const std::string& sid) {
   return "";
 }
 
+// Given a SID, retrieve information about the matching user
+void genUser(const std::string& sidString, QueryData& results) {
+  Row r;
+
+  r["uuid"] = sidString;
+  r["directory"] = getUserHomeDir(sidString);
+
+  PSID sid;
+  auto ret = ConvertStringSidToSidA(sidString.c_str(), &sid);
+  if (ret == 0) {
+    VLOG(1) << "Convert SID to string failed with " << GetLastError();
+  }
```

Comment:
Early return here?

Reply:
Ah, not only that, but the error message is flipped. It's attempting to convert a string to a SID, not vice versa. Rather than return early, why don't I skip what can't be resolved, but return the row containing partial information.